plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,922 @@
+
+  # --- Template for MultiGateway ---
+  multigateway:
+    replicas: 2
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multigateway
+            topologyKey: "kubernetes.io/hostname"
+        - weight: 50
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multigateway
+            topologyKey: "topology.kubernetes.io/zone"
+    resources:
+      requests:
+        cpu: "500m"
+        memory: "512Mi"
+      limits:
+        cpu: "1"
+        memory: "1Gi"
+
+  # --- Template for MultiAdmin ---
+  multiadmin:
+    replicas: 1
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multiadmin
+            topologyKey: "kubernetes.io/hostname"
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+      limits:
+        cpu: "200m"
+        memory: "256Mi"
+
+  managedTopoServer:
+    image: quay.io/coreos/etcd:v3.5.17
+    imagePullPolicy: "IfNotPresent"
+    imagePullSecrets:
+      - name: "my-registry-secret"
+    replicas: 3
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: topo-server
+            topologyKey: "kubernetes.io/hostname"
+        - weight: 50
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: topo-server
+            topologyKey: "topology.kubernetes.io/zone"
+    dataVolumeClaimTemplate:
+      accessModes: ["ReadWriteOnce"]
+      resources:
+        requests:
+          storage: "10Gi"
+status:
+  # This list is populated by the MultigresCluster controllers using it.
+  # It shows which clusters are actively referencing this template.
+  consumers:
+    - name: "example-multigres-cluster"
+      namespace: "example"
+    - name: "other-cluster"
+      namespace: "default"
+```
+
+
+
+## TopoServer CR - Read-Only Child of MultigresCluster (or cell if localtopology) 
+
+ * This CR applies to both the global topology and the local topology server. It uses the same CRD for both.
+      * If global, it's directly owned by the `MultigresCluster`.
+      * If local, it's owned by the `Cell`.
+  * This CR does not exist if the user configures an external etcd server.
+  * The `Cell` uses the global `TopoServer` by default if a local one is not configured.
+  * Because this is considered a separate resource, the image is declared locally within it, not globally.
```

Comment:
What does this mean? My understanding is that the image definitions are going to be propagated to each child resource anyways, as they won't have any reference to the template?

Reply:
The template for the toposerver also includes the image. It is treated as a separate resource so the config includes this as well.